frontend/src/hooks.ts

```diff
@@ -1,6 +1,44 @@
 import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux'
 import type { RootState, AppDispatch } from './store'
+import { FMSId } from './models/fms_api_models'
+import { useEffect, useMemo, useState } from 'react'
+import { fetchLibrariesForSamples, fetchSamples } from './modules/cache/cache'
+import { selectLibrariesByID, selectSamplesByID } from './selectors'
+import { SampleAndLibrary } from './components/WorkflowSamplesTable/ColumnSets'
 
 // Use throughout your app instead of plain `useDispatch` and `useSelector`
 export const useAppDispatch: () => AppDispatch = useDispatch
-export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
\ No newline at end of file
+export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
+
+export function useSampleList(sampleIDs: FMSId[], offset?: number, size?: number) {
```

Comment:
i personally like more precise variable names ... size would be clearer as page_size ... (at least to me). Up to you ...